src/errors/BaseError.ts

```diff
@@ -0,0 +1,32 @@
+import { isNullOrEmpty } from '../utils/stringUtils';
+import { ErrorProperties } from './types';
+
+/**
+ * Base Error class is intended to be used as a base class for every type of Error generated
+ * @param message an error message in human readible format
+ * @param exception a generated exception provided as a {Record<string, unknown>} type
+ * @export
+ * @class BaseError
+ * @extends {Error}
+ */
+export default class BaseError extends Error {
+    protected properties: ErrorProperties;
+    hasBeenLogged: boolean;
+
+    constructor(message: string, exception: Record<string, unknown>) {
```

Comment:
Exception should be optional, you can take a look in echopedia where this already has been done